### PR TITLE
assertMacroExpansion should assert if there are errors

### PR DESCRIPTION
**Pull request by bnbarham**
Invalid macro expansions should generate an assertion in `assertMacroExpansion` rather than being ignored.

### CODE REVIEW

**Comment by bnbarham** [comment]
@swift-ci please test

**Review by rintaro — APPROVED** [review]

**Review by kimdv — APPROVED** [review]

File: Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift [hunk 1]
```diff
@@ -272,10 +272,13 @@ public func assertMacroExpansion(
   let context = BasicMacroExpansionContext(
     sourceFiles: [origSourceFile: .init(moduleName: testModuleName, fullFilePath: testFileName)]
   )
-  let expandedSourceFile = origSourceFile.expand(macros: macros, in: context).formatted(using: BasicFormat(indentationWidth: indentationWidth))
+  let expandedSourceFile = origSourceFile.expand(macros: macros, in: context)
+  let formattedSourceFile = expandedSourceFile.formatted(using: BasicFormat(indentationWidth: indentationWidth))
+
+  XCTAssertFalse(expandedSourceFile.hasError, "Expanded source should not contain any syntax errors")
```
Comment (ahoppen) [inline Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift:278]: What do you think about running `ParseDiagnsoticGenerator` to actually show the syntax errors?

Comment (bnbarham) [reply Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift]: I added this but IMO they still don't make much sense without the syntax tree. So I added that in as well.

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by bnbarham** [comment]
@swift-ci please test

**Review by ahoppen — APPROVED** [review]

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**